File: algo-bubble-sort/js/bubbleSort.js
// Your Code Here
const bubbleSort = (array) => {
    var swaps = 0;
    var iterations = 0;
    const n = array.length;
    for (let i = 0; i < n; i++) {
        let swapped = false;
        for (let j = 0; j < n; j++) {
            iterations++;
            if (array[j] < array[j - 1]) {
                let temp = array[j];
                array[j] = array[j - 1];
                array[j - 1] = temp;
                swapped = true;
                swaps++;
            }
        }
        if (!swapped) {
            break;
        }
    }
    console.log("Swaps: ", swaps);
    console.log("Iterations: ", iterations);
    return array;
};


var sequence = [4, 3, 5, 0, 1, 6];
console.log("Final Result: ", (bubbleSort(sequence)));

File: algo-bubble-sort/python/bubble_sort.py
# from msilib import sequence


def bubble_sort(list):
    swaps = 0
    n = len(list)
    iteration_count = 0
    for _ in range(n):
        swapped = False
        for index in range(1, len(list)):
            iteration_count += 1
            if list[index] < list[index - 1]:
                list[index - 1], list[index] = list[index], list[index - 1]
                swaps += 1
                swapped = True
        if not swapped:
            break


    print(f"Swaps: {swaps}")
    print(f"Iterations: {iteration_count}")
    return list

sequence = [4, 3, 5, 0, 1, 6]
print(f"Final Result: {bubble_sort(sequence)}")